projects/ngx-datatable/src/lib/utils/math.ts

```diff
@@ -75,6 +75,29 @@ function scaleColumns(colsByGroup: any, maxWidth: any, totalFlexGrow: any) {
       }
     }
   } while (remainingWidth !== 0);
+
+  // Adjust for any remaining offset in computed widths vs maxWidth by tweaking the biggest column
+  let totalWidthAchieved = 0;
+  const biggestColumnRef = {
+    width: 0,
+    ref: null
+  };
+
+  for (const attr in colsByGroup) {
+    if (colsByGroup.hasOwnProperty(attr)) {
+      for (const column of colsByGroup[attr]) {
+        totalWidthAchieved += column.width;
+
+        if (column.width > biggestColumnRef.width) {
```

Comment:
```suggestion
        if (column.canAutoResize && column.width > biggestColumnRef.width) {
```

Reply:
Unless I missed something, the previous implementation exceeded maxWidth in case `sum(minWidth) > maxWidth`.
I think we should keep this behavior.

Reply:
Ok, I've added checks for that, also refactored the logic a bit, given that if we violate minWidth on the biggest column, we may apply the correction on one of the other columns. I've also added two more unit tests, both with edge cases.